sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -313,131 +112,254 @@ object ScalaNativePluginInternal {
     },
     addCompilerPlugin(
       "org.scala-native" % "nscplugin" % nativeVersion cross CrossVersion.full),
-    nativeSharedLibrary := false,
-    nativeClang := None,
-    nativeClangPP := None,
+    nativeClang := {
```

Comment:
Actually, as tasks, they will get evaluated only if another task depending on them is executed. So it should be quite fine for #562.